scripts/check.sh

```diff
@@ -163,25 +163,17 @@ if [[ "${VERBOSE}" == true ]]; then
   env | egrep '^TRAVIS_(BRANCH|COMMIT|PULL)' | sort || true
 fi
 
-# When travis clones a repo for building, it uses a shallow clone. When
-# building a branch it can sometimes give a revision range that refers to
-# commits that don't exist in the shallow clone. This has been observed in a
-# branch build where the branch only has a single commit. The cause of this
-# behavior is unclear but as a workaround ...
-if [[ "${START_REVISION}" == *..* ]]; then
-  RANGE_START="${START_REVISION/..*/}"
-  RANGE_END="${START_REVISION/*../}"
+# When travis clones a repo for building, it uses a shallow clone. After the
+# first commit on a non-master branch, TRAVIS_COMMIT_RANGE is not set and
+# START_REVISION is "master" instead of a range.
 
+if [[ -z "${TRAVIS_COMMIT_RANGE}" ]]; then
```

Comment:
Depending on travis values like this makes it harder to test locally, and duplicates the fact that we already take this value on the command-line.

A better fix would be to ensure that the default `START_REVISION` actually exists so that `git rev-parse` doesn't fail.

In this case we're going out of our way to ensure that `origin/master` exists, but didn't create a local tracking branch for it. Shouldn't the fix be to ensure that we handle the case where there's no local tracking branch?

That is, isn't the fix just to change the default for `START_REVISION` to `origin/master`?

Reply:
Done. And verified at https://travis-ci.org/firebase/firebase-ios-sdk/builds/533528751

This will change local behavior for master being out of sync with origin/master, but maybe ok?

Reply:
If `master` is out of sync with `origin/master`, we likely prefer the latter anyway.